master/internal/resourcemanagers/kubernetes/spec.go

```diff
@@ -365,7 +365,21 @@ func (p *pod) createPodSpec(ctx *actor.Context, scheduler string) error {
 		p.ports = append(p.ports, port)
 	}
 
-	envVars, err := p.configureEnvVars(spec.EnvVars(), env, deviceType)
+	specEnvVars := spec.EnvVars()
+	var specEnvFrom []k8sV1.EnvFromSource
+	podSpec := env.PodSpec()
+	if podSpec != nil {
+		for _, container := range podSpec.Spec.Containers {
+			if container.Name == "determined-container" {
+				for _, e := range container.Env {
+					specEnvVars[e.Name] = e.Value
+				}
+				specEnvFrom = container.EnvFrom
+			}
+		}
+	}
```

Comment:
why is this not all in `configureEnvVars`? or, better even (? maybe), it seems like it could be merged in `configurePodSpec`, where stuff for the `determinedContainer` is merged into it from the default pod spec (where requests/limits are merged in). there are a few patterns here, it's a bit confusing to figure out where stuff happens and why i think. why this way?

Reply:
Good points. I think my original thinking was that I should modify the env before the container gets constructed, but you're right - it does look like `configurePodSpec` is a much more natural place to put it.